Taskfile.yaml

```diff
@@ -382,6 +382,9 @@ tasks:
     dir: prqlc/bindings/php
     cmds:
       - vendor/bin/phpunit tests
+
+  add-safe-directory:
+    - git config --global --add safe.directory /workspaces/prql
```

Comment:
I do not think this should be included in the Taskfile. There are very limited situations where this absolute path is valid.

Reply:
I see the difficulty. _/workspaces/prql_ works fine for the Dev Container, but is unlikely to be valid in most other situations.

How could this be modified?

(I suspect this bites 100% of people who work on PRQL. For those who compile natively, it's a one-time fix; for Dev Container folks (like me) I have to re-run this command each time the container gets rebuilt...)

Reply:
I tend to agree with the downside, but where should it go?

(and do you happen to know why we get this problem?)

Reply:
I have not seen this problem occur, and if it does occur, it is most likely a problem with the devcontainer CLI and should not be addressed in this repository.
I suggest reverting this change.

Reply:
> (and do you happen to know why we get this problem?)

Maybe this issue microsoft/vscode-remote-release#8656

Reply:
Would putting this in the Dockerfile rather than the Taskfile work?

[Here](https://github.com/codespell-project/codespell/pull/3361/files) they have a script that runs on install and adds the command, which I think would be similar to having it in the Dockerfile?

Reply:
Yes - Their fix is substantially the same as this PR. Their `postCreateCommand` calls a _post\_create.sh_ script in the _.devcontainer_ directory that executes the `git commit...` command.

In the PRQL PR (#4830), the `postCreateCommand` calls the `git commit...` command via `task`. Since that task would only be called from the Dev Container setup, it's not dangerous, but it could still be a foot-gun for people who  only look at the _Taskfile.yaml_.

I see that @eitsupi has reverted this - I am OK with this. I plan to see if I can explain the behavior (or make it occur elsewhere...) If I don't find a way to avoid it, I might add the _post\_create.sh_ script to keep that command local to the Dev Container environment. 

I'll write more as I find more. Thanks

Reply:
@eitsupi would you be OK with this is going into the Dockerfile? I had thought your objection was about it being in the Taskfile, but lmk if I'm misinterpreting...

Reply:
I am quite skeptical about the conditions under which this problem arises in the first place.
You see, I have been using Dev Containers for years and have never encountered this problem (of course, it was only relatively recently, a few years ago, when Git security was changed and this kind of error started occurring).

But my suspicion is that it is a user manipulation issue and that setting this up for all containers is the wrong direction.
At the very least, we should be very aware that this is an issue that should be resolved upstream and reported upstream.

Reply:
We're not the only repo which has this problem — my link above was: https://github.com/codespell-project/codespell/pull/3361/files...

I agree with the principle of solving the root issue rather than adding our patches — but adding a small patch with a `TODO: not sure why we need this, but we get error $foo without it` close to the issue seems like a reasonable balance, no?

Reply:
OK, let's wait to hear from the upstream issue then. Thanks.

Reply:
Let's move this conversation to https://github.com/PRQL/prql/pull/4836#issuecomment-2296755956 which has much more context